src/build/arg/mod.rs

```diff
@@ -1917,11 +1918,14 @@ impl<'help> Arg<'help> {
     /// [`Result`]: https://doc.rust-lang.org/std/result/enum.Result.html
     /// [`Err(String)`]: https://doc.rust-lang.org/std/result/enum.Result.html#variant.Err
     /// [`Rc`]: https://doc.rust-lang.org/std/rc/struct.Rc.html
-    pub fn validator_os<F, O>(mut self, mut f: F) -> Self
+    pub fn validator_os<F, O, E>(mut self, mut f: F) -> Self
     where
-        F: FnMut(&OsStr) -> Result<O, String> + Send + 'help,
+        F: FnMut(&OsStr) -> Result<O, E> + Send + 'help,
+        E: Into<Box<dyn Error + Sync + Send + 'static>>,
```

Comment:
Please use `Send + Sync` order instead of the reverse you were using in this file. I know it doesn't matter, but it kept tripping me up while I am reading the code since I am used to it in that way.